raiden/encoding/format.py

```diff
@@ -127,19 +145,32 @@ def __setattr__(self, name, value):
                 pad_value = b'\x00' * pad_size
                 value = pad_value + value
 
-            self.data[slice_] = value
+            data = object.__getattribute__(self, 'data')
+            data[slice_] = value
         else:
             super(self.__class__, self).__setattr__(name, value)
 
+    def __repr__(self):
+        return '<{} [...]>'.format(buffer_name)
+
+    def __len__(self):
+        return size
+
+    def __dir__(self):
+        return sorted_names
+
     attributes = {
```

Comment:
Yes, agreed, didn't want to refactor this now, I only needed to expose the fields slices to write the tests.